### ISSUE TITLE
Ambiguity warnings in 0.4

### ISSUE THREAD

**Issue by lobingera** [JuliaMath/FixedPointNumbers.jl#77]
#55 was closed in sept 2016, however i see the same with 
```
               _
   _       _ _(_)_     |  A fresh approach to technical computing
  (_)     | (_) (_)    |  Documentation: http://docs.julialang.org
   _ _   _| |_  __ _   |  Type "?help" for help.
  | | | | | | |/ _` |  |
  | | |_| | | | (_| |  |  Version 0.4.5 (2016-03-18 00:58 UTC)
 _/ |\__'_|_|_|\__'_|  |  
|__/                   |  x86_64-linux-gnu

julia> Pkg.init()
INFO: Initializing package repository /home/lobi/.julia/v0.4
INFO: Cloning METADATA from git://github.com/JuliaLang/METADATA.jl

julia> Pkg.add("Cairo")
INFO: Installing BinDeps v0.4.5
INFO: Installing Cairo v0.2.35
INFO: Installing ColorTypes v0.2.12
INFO: Installing Colors v0.6.9
INFO: Installing Compat v0.19.0
INFO: Installing FixedPointNumbers v0.2.1
INFO: Installing Graphics v0.1.3
INFO: Installing Reexport v0.0.3
INFO: Installing SHA v0.3.1
INFO: Installing URIParser v0.1.8
INFO: Building Cairo
INFO: Recompiling stale cache file /home/lobi/.julia/lib/v0.4/BinDeps.ji for module BinDeps.
INFO: Recompiling stale cache file /home/lobi/.julia/lib/v0.4/Compat.ji for module Compat.
INFO: Recompiling stale cache file /home/lobi/.julia/lib/v0.4/URIParser.ji for module URIParser.
INFO: Recompiling stale cache file /home/lobi/.julia/lib/v0.4/SHA.ji for module SHA.
INFO: Package database updated

julia> Pkg.test("Cairo")
INFO: Testing Cairo
INFO: Recompiling stale cache file /home/lobi/.julia/lib/v0.4/Cairo.ji for module Cairo.
INFO: Recompiling stale cache file /home/lobi/.julia/lib/v0.4/Colors.ji for module Colors.
INFO: Recompiling stale cache file /home/lobi/.julia/lib/v0.4/FixedPointNumbers.ji for module FixedPointNumbers.
WARNING: New definition 
    floattype(Type{#T<:FixedPointNumbers.Fixed}) at /home/lobi/.julia/v0.4/FixedPointNumbers/src/fixed.jl:16
is ambiguous with: 
    floattype(Type{FixedPointNumbers.FixedPoint{#T<:Union{Int8, UInt16, Int16, UInt8}, #f<:Any}}) at /home/lobi/.julia/v0.4/FixedPointNumbers/src/FixedPointNumbers.jl:89.
To fix, define 
    floattype(Type{FixedPointNumbers.Fixed{_<:Union{Int8, Int16}, #f<:Any}})
before the new definition.
WARNING: New definition 
    floattype(Type{#T<:FixedPointNumbers.UFixed}) at /home/lobi/.julia/v0.4/FixedPointNumbers/src/ufixed.jl:14
is ambiguous with: 
    floattype(Type{FixedPointNumbers.FixedPoint{#T<:Union{Int8, UInt16, Int16, UInt8}, #f<:Any}}) at /home/lobi/.julia/v0.4/FixedPointNumbers/src/FixedPointNumbers.jl:89.
To fix, define 
    floattype(Type{FixedPointNumbers.UFixed{_<:Union{UInt16, UInt8}, #f<:Any}})
before the new definition.
INFO: Recompiling stale cache file /home/lobi/.julia/lib/v0.4/ColorTypes.ji for module ColorTypes.
INFO: Recompiling stale cache file /home/lobi/.julia/lib/v0.4/Reexport.ji for module Reexport.
INFO: Recompiling stale cache file /home/lobi/.julia/lib/v0.4/Graphics.ji for module Graphics.
INFO: Cairo tests passed

julia> Pkg.status()
1 required packages:
 - Cairo                         0.2.35
9 additional packages:
 - BinDeps                       0.4.5
 - ColorTypes                    0.2.12
 - Colors                        0.6.9
 - Compat                        0.19.0
 - FixedPointNumbers             0.2.1
 - Graphics                      0.1.3
 - Reexport                      0.0.3
 - SHA                           0.3.1
 - URIParser                     0.1.8

```

**Comment by timholy** [comment]
Check out the version number of FixedPointNumbers that got installed and then compare the current release as listed under the repo's "releases" tab.

**Comment by lobingera** [comment]
I'm somehow aware that there are newer versions, however on the same setup (note: above was starting with a fresh Pkg.init) my Pkg.update tells me:
```
julia> Pkg.update()
INFO: Updating METADATA...
INFO: Updating cache of BinDeps...
INFO: Updating Compat...
INFO: Updating Cairo...
INFO: Computing changes...
INFO: No packages to install, update or remove
```
I don't see this a usability problem or similar, but still, most of automatic testing works like this an will produce warnings.
Do you see this a Pkg + REQUIRE problem?

**Comment by timholy** [comment]
Looking more closely, those are completely different ambiguity warnings than the ones in #55. I see that you're using 0.4, and the latest releases all require 0.5. If they bother you, you might be able to fix them by applying the same change as in #75; we could merge a PR into a separate branch and we could do another release in the v0.2 cycle.

**Comment by lobingera** [comment]
> If they bother you

I see this here as reporting, only. I need to solve some other 0.4 compatibility problems for my Cairo testing update and this showed up along that.
